### PR TITLE
feat: EPAS support in web-component

### PR DESCRIPTION
**Pull request by birme**
This PR addresses #43 and adds support for EPAS tracking in web-component

### CODE REVIEW

File: packages/web-player-component/README.md [hunk 1]
```diff
@@ -10,6 +10,8 @@ This web component is powered by the Eyevinn WebPlayer and can be used on an HTM
 | `starttime`              | seconds  | Specifies where in the timeline to start playing the stream        |
 | `autoplay`               | autoplay | Specifies that the video will start playing as soon as it is ready |
 | `muted`                  | muted    | Specifies that the audio output of the video should be muted       |
+| `autoplay-visible`       | autoplay-visible | Specifies that the video will start playing as soon as it is ready and only when visible (pauses when not visible) |
```
Comment (slowmove) [inline packages/web-player-component/README.md:13]: 👍

**Review by slowmove — COMMENTED** [review]
It's a little bit chaotic with the nestling of using promises as is, without async await - though ok of course.

We should probably add the methods to report error and bitrate though?

**Comment by birme** [comment]
> It's a little bit chaotic with the nestling of using promises as is, without async await - though ok of course.
> 
Yeah, I agree and was wondering if I should rewrite it. Was unsure if it was intentionally using promises instead of async / await but I cannot come up with a good reason.

> We should probably add the methods to report error and bitrate though?
Yes, right. Of course.

**Comment by birme** [comment]
Did some refactoring, added bitrate and error reporting and I also believe I have solved #38 as well. Updated test html to cover all cases.

File: packages/web-player-component/src/index.js [hunk 2]
```diff
@@ -36,31 +37,85 @@ export default class PlayerComponent extends HTMLElement {
...
+        });
+      } catch (err) {
+        console.error(err);
+        this.playerAnalytics.deinit();
+        this.playerAnalytics = null;
+      }
+    }
+  }
+
+  async analyticsLoad() {
```
Comment (slowmove) [inline packages/web-player-component/src/index.js:88]: Nothing asynchronous in this method right?

File: packages/web-player-component/src/index.js [hunk 3]
```diff
@@ -36,31 +37,85 @@ export default class PlayerComponent extends HTMLElement {
...
-          if (this.hasAttribute('starttime')) {
-            this.video.currentTime = this.getAttribute('starttime');
-          }
-          if (this.hasAttribute('autoplay')) {
-            this.player.play();
-          }
-        });
+        await this.analyticsInit();
+        await this.player.load(this.getAttribute('source'));
+        await this.analyticsLoad();
```
Comment (slowmove) [inline packages/web-player-component/src/index.js:103]: Nothing asynchronous in `analyticsLoad` that we need to wait for, right?

File: packages/web-player-component/src/index.js [hunk 4]
```diff
@@ -36,31 +37,85 @@ export default class PlayerComponent extends HTMLElement {
...
       }
     }
     if (name === 'muted') {
-      if (this.hasAttribute("muted")) {
-        this.video.muted = true;
-      }
-      else {
-        this.video.muted = false;
-      }
+      this.video.muted = this.hasAttribute('muted');
```
Comment (slowmove) [inline packages/web-player-component/src/index.js:118]: Maybe `!!this.hasAttribute('muted');` 
( `!!` cast to boolean)

**Review by slowmove — COMMENTED** [review]
No big things. (see comments)

I do as well rather like to return early
i.e. `if (!this.playerAnalytics) return;` rather than wrap and add indentation.
Though that is just an opinion.
https://medium.com/swlh/return-early-pattern-3d18a41bba8

File: packages/web-player-component/src/index.js (hunk 2 above)
Comment (birme) [reply packages/web-player-component/src/index.js:88]: That's correct, I'll fix

**Comment by birme** [comment]
> No big things. (see comments)
> 
> I do as well rather like to return early i.e. `if (!this.playerAnalytics) return;` rather than wrap and add indentation. Though that is just an opinion. https://medium.com/swlh/return-early-pattern-3d18a41bba8

Ah, no problem. Will fix.

**Review by slowmove — APPROVED** [review]

**Pull request merged**